ocaml/typing/includemod.ml

```diff
@@ -471,6 +446,33 @@ module Sign_diff = struct
     }
 end
 
+(* Quickly compare module types without expanding them *)
+let rec shallow_modtypes env subst mty1 mty2 =
```

Comment:
No, it's an inclusion check but it only succeeds for (some) inclusions which don't require a coercion. For instance, `S with M <= S` (as the comment says) and that never needs a coercion (since for that to be type correct, `M : S` must be true).

Reply:
Expanded the comment.